File: Earthshake/EarthshakeModelFactory.m
//
//  EarthshakeBuilderFactory.m
//  Earthshake
//
//  Created by Andrey Paunov on 2016-03-26.
//  Copyright © 2016 Andrey Paunov. All rights reserved.
//

#import "EarthshakeModelFactory.h"

@implementation EarthshakeModelFactory

- (EarthshakeItemBuilder *) createEarthshakeItemBuilder
{
    if (self.earthshakeItemBuilder)
    {
        return self.earthshakeItemBuilder;
    }

    self.earthshakeItemBuilder = [[EarthshakeItemBuilder alloc] init];

    return self.earthshakeItemBuilder;
}

@end

File: Earthshake/EarthshakeModelFactory.swift
//
//  EarthshakeModelFactory.swift
//  Earthshake
//
//  Created by Andrey on 2016-12-08.
//  Copyright © 2016 AP. All rights reserved.
//

import Foundation

class EarthshakeModelFactory : NSObject
{
    var earthshakeItemBuilder: EarthshakeItemBuilder?
    
    func createEarthshakeItemBuilder() -> EarthshakeItemBuilder
    {
        if (earthshakeItemBuilder != nil)
        {
            return earthshakeItemBuilder!
        }

        earthshakeItemBuilder = EarthshakeItemBuilder()

        return earthshakeItemBuilder!
    }
}
